crates/sui-storage/src/lock_service.rs

```diff
@@ -174,7 +170,17 @@ impl LockServiceImpl {
             Some(prev_seq) => Ok(prev_seq),
 
             None => {
-                // Assert that the tx locks exist
+                // Assert that the tx locks exist.
+                //
+                // Note that the locks may not be set to this particular tx:
+                //
+                // 1. Lock existence prevents re-execution of old certs when objects have been
+                //    upgraded
+                // 2. Not all validators lock, just 2f+1, so transaction should proceed regardless
```

Comment:
Thanks for adding.  I was trying to think of any scenarios for which this wouldn't work.   Think this is fine so long as genesis and new object creation initialize the lock properly.